### PR TITLE
feat: P4ADEV-126 Supported connection between a certificate KV & CDN

### PR DESCRIPTION
**Pull request by umbcoppolabottazzi**
### List of changes

With this PR, the association between the certificate stored on KV and the CDN is supported in cases where the hostname does not match the APEX domain

### Type of changes

- [ ] Add new module
- [x] Update existing module
- [ ] Remove existing module

### Does this introduce a breaking change?

- [ ] Yes
- [x] No

### Run checks

Useful commands to run checks on local machine

```sh
bash .utils/terraform_run_all.sh init local
pre-commit run -a
```

### CODE REVIEW

File: cdn/variables.tf [hunk 1]
```diff
@@ -321,6 +331,24 @@ variable "keyvault_vault_name" {
...
+  default     = null
+}
+
+variable "azuread_service_principal_azure_cdn_frontdoor_id" {
+  type        = string
+  description = "Azure CDN Front Door Principal ID - Microsoft.AzureFrontDoor-Cdn"
+  default     = null
+}
+
+variable "custom_hostname_kv" {
```
Comment (diegolagospagopa) [inline cdn/variables.tf:346]: Please change into

```suggestion
variable "custom_hostname_kv_enabled" {
```

File: cdn/variables.tf [hunk 2]
```diff
@@ -334,6 +362,11 @@ variable "advanced_threat_protection_enabled" {
   default = false
 }
 
+variable "enable_resource_advanced_threat_protection" {
```
Comment (diegolagospagopa) [inline cdn/variables.tf:365]:
```suggestion
variable "resource_advanced_threat_protection_enabled" {
```

File: cdn/variables.tf [hunk 3]
```diff
@@ -29,6 +34,11 @@ variable "location" {
   type = string
 }
 
+variable "cdn_location" {
+  type        = string
+  default     = null
+  description = "Specific location for CDN"
```
Comment (diegolagospagopa) [inline cdn/variables.tf:40]: Please put some examples

File: cdn/variables.tf [hunk 4]
```diff
@@ -5,6 +5,11 @@ variable "name" {
   type = string
 }
 
+variable "tenant_id" {
+  type    = string
+  default = null
```
Comment (diegolagospagopa) [inline cdn/variables.tf:10]: If is mandatory, remove default

**Review by diegolagospagopa — CHANGES_REQUESTED** [review]

File: cdn/variables.tf (hunk 4 above)
Comment (umbcoppolabottazzi) [reply cdn/variables.tf:10]: This variable is optional

File: cdn/variables.tf (hunk 3 above)
Comment (umbcoppolabottazzi) [reply cdn/variables.tf]: - Added a clearer description
- Added an example with a different location in the test folder

Thanks

**Review by diegolagospagopa — APPROVED** [review]

**Pull request merged**